### PR TITLE
Mule 15872: Using splitter aggregator implicitly inside a request-reply scope wrapped with for each does not return the results back intermittently.

### PR DESCRIPTION
**Pull request by emariotti3**

### CODE REVIEW

File: core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java [hunk 1]
```diff
@@ -50,6 +50,7 @@
     protected MessageInfoMapping messageInfoMapping;
     protected int batchSize;
     protected String counterVariableName;
+    private static int proccessedSequences = 0;
```
Comment (mbuchwald) [inline core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java]: Not sure why this should be `static`, but probably need to be `AtomicInteger`

File: pom.xml [hunk 2]
```diff
@@ -144,7 +144,7 @@
         <antlrVersion>3.5.2</antlrVersion>
         <asmVersion>6.2</asmVersion>
         <aspectjVersion>1.9.1</aspectjVersion>
-        <asyncHttpClientVersion>1.14-MULE-007</asyncHttpClientVersion>
+        <asyncHttpClientVersion>1.14-MULE-008-SNAPSHOT</asyncHttpClientVersion>
```
Comment (mbuchwald) [inline pom.xml]: Why is this change here?

File: tests/integration/src/test/java/org/mule/test/usecases/routing/response/VMRequestReplyInForEachTestCase.java [hunk 3]
```diff
@@ -6,13 +6,70 @@
...
+
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+
+import org.junit.Test;
+
 public class VMRequestReplyInForEachTestCase extends RequestReplyInForEachTestCase
 {
+    protected static final int TIMEOUT = 5000;
```
Comment (mbuchwald) [inline tests/integration/src/test/java/org/mule/test/usecases/routing/response/VMRequestReplyInForEachTestCase.java]: If the other property is now `protected`, this shouldn't be necessary (or the other one could remain `private`)

File: core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java [hunk 4]
```diff
@@ -146,7 +150,7 @@ public Object transform(Object input)
                 if ((!correlationSet && (enableCorrelation == CorrelationMode.IF_NOT_SET))
                     || (enableCorrelation == CorrelationMode.ALWAYS))
                 {
-                    message.setCorrelationId(correlationId);
+                    message.setCorrelationId(correlationId + processedSequences.get());
```
Comment (elrodro83) [inline core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java]: put an `_` in the middle to separate the original id from the suffix

File: core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java [hunk 5]
```diff
@@ -50,6 +51,7 @@
     protected MessageInfoMapping messageInfoMapping;
     protected int batchSize;
     protected String counterVariableName;
+    private static AtomicInteger processedSequences = new AtomicInteger(0);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java]: does this need to be at the class level and atomic? Can't it be scoped to the `process` method.

File: tests/integration/src/test/java/org/mule/test/usecases/routing/response/VMRequestReplyInForEachTestCase.java [hunk 6]
```diff
@@ -6,13 +6,68 @@
...
         return "org/mule/test/usecases/routing/response/vm-request-reply-in-for-each.xml";
     }
 
+    @Override
+    public void setUp() throws Exception
+    {
+        super.setUp();
+        collectionValues.add(new ArrayList<String>());
+        collectionValues.add(new ArrayList<String>());
+        collectionValues.get(0).add("value1");
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/usecases/routing/response/VMRequestReplyInForEachTestCase.java]: Use an immutable list builder from guava.

File: core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java [hunk 7]
```diff
@@ -51,7 +50,7 @@
     protected MessageInfoMapping messageInfoMapping;
     protected int batchSize;
     protected String counterVariableName;
-    private static AtomicInteger processedSequences = new AtomicInteger(0);
+    protected boolean sequential = false;
```
Comment (elrodro83) [inline core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java]: make final

File: core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java [hunk 8]
```diff
@@ -150,7 +146,8 @@ public Object transform(Object input)
                 if ((!correlationSet && (enableCorrelation == CorrelationMode.IF_NOT_SET))
                     || (enableCorrelation == CorrelationMode.ALWAYS))
                 {
-                    message.setCorrelationId(correlationId + processedSequences.get());
+                    String suffix = sequential? ("-" + correlationSequence) : "";
```
Comment (elrodro83) [inline core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java]: format, and inline

File: core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java [hunk 9]
```diff
@@ -87,6 +86,8 @@ protected boolean isSplitRequired(MuleEvent event)
         return true;
     }
 
+    protected boolean isSequential() { return false; }
```
Comment (elrodro83) [inline core/src/main/java/org/mule/routing/outbound/AbstractMessageSequenceSplitter.java]: format
